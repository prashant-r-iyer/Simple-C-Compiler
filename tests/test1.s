	.text

.globl main
main:
	# Save Frame pointer
	pushq %rbp
	movq %rsp, %rbp
	subq $256, %rsp
	pushq %rbx
	pushq %rbx
	pushq %r10
	pushq %r13
	pushq %r14
	pushq %r15
	pushq %r11
	pushq %r9
	movq $string0, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

	# Restore registers
	popq %r9
	popq %r11
	popq %r15
	popq %r14
	popq %r13
	popq %r10
	popq %rbx
	popq %rbx
	addq $256, %rsp

	leave
	ret

string0:
	.string "Hello world...\n"

