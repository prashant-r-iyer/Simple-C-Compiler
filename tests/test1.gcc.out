test1.c: In function ‘main’:
test1.c:4:9: warning: implicit declaration of function ‘printf’ [-Wimplicit-function-declaration]
    4 |         printf("Hello world...\n");
      |         ^~~~~~
test1.c:1:1: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
  +++ |+#include <stdio.h>
    1 | 
test1.c:4:9: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
    4 |         printf("Hello world...\n");
      |         ^~~~~~
test1.c:4:9: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
